frontend/public/components/secrets/create-secret.tsx

```diff
@@ -139,6 +160,211 @@ const withSecretForm = (SubForm) => class SecretFormComponent extends React.Comp
+          email: this.state.email
+        }
+      };
+      this.props.onChange(this.state.isNewFormat ? {[AUTHS_SYMBOL]: secretData} : secretData);
+    }
+  }
+  render() {
+    return <React.Fragment>
+      <div className="form-group">
+        <label className="control-label">Registry Server Address</label>
+        <div>
+          <input className="form-control"
+            id="address"
+            type="text"
+            name="address"
+            onChange={this.changeData}
+            value={this.state.address}
+            required />
+        </div>
+      </div>
+      <div className="form-group">
+        <label className="control-label">Username</label>
+        <div>
+          <input className="form-control"
+            id="username"
+            type="text"
+            name="username"
+            onChange={this.changeData}
+            value={this.state.username}
+            required />
+        </div>
+      </div>
+      <div className="form-group">
+        <label className="control-label">Password</label>
+        <div>
+          <input className="form-control"
+            id="password"
+            type="password"
+            name="password"
+            onChange={this.changeData}
+            value={this.state.password}
+            required />
+        </div>
+      </div>
+      <div className="form-group">
+        <label className="control-label">Email</label>
+        <div>
+          <input className="form-control"
+            id="email"
+            type="text"
+            name="email"
+            onChange={this.changeData}
+            value={this.state.email}
+            required />
+        </div>
+      </div>
+    </React.Fragment>;
+  }
+}
+
+class UploadConfigSubform extends React.Component<UploadConfigSubformProps, UploadConfigSubformState> {
+  constructor (props) {
+    super(props);
+    this.state = {
+      configFile: _.isEmpty(this.props.stringData) ? '' : JSON.stringify(this.props.stringData),
+      parseError: false,
+    };
+    this.changeData = this.changeData.bind(this);
+    this.onFileChange = this.onFileChange.bind(this);
+  }
+  changeData(event) {
+    this.updateState(_.attempt(JSON.parse, event.target.value), event.target.value);
+  }
+  onFileChange(fileData) {
+    this.updateState(_.attempt(JSON.parse, fileData), fileData);
+  }
+  updateState(parsedData, stringData) {
+    this.setState({
+      configFile: stringData,
+      parseError: _.isError(parsedData),
+    }, () => !_.isError(parsedData) && this.props.onChange(parsedData));
+  }
+  render() {
+    return <div className="form-group">
+      <label className="control-label" htmlFor="docker-config">Configuration File</label>
+      <div>
+        <FileInput onChange={this.onFileChange} />
+        <p className="help-block">Upload a .dockercfg or .docker/config.json file</p>
+        <textarea className="form-control co-create-secret-form__textarea"
+          id="docker-config"
```

Comment:
I haven't looked at this a ton in this console, but in the old `angular` console we would occasionally have errors due to duplicate `id` attribs.  Should this generate a unique ID each time the form is rendered to the page?  Do we care that the `name` and `id` are following different conventions?

Reply:
Shouldn't be an issue unless the form is rendered twice on the same page. I'd rather not add the logic until we need it.

Reply:
I wouldn't hold a PR over it, but I tend to think that if we use an `id` we should always ensure it is unique. I'm not sure it hurts anything, and the logic is pretty minimal via `_.uniqueId('prefix-')`.  I can see where it may get a touch more complicated if the `id` needs to be created in `componentWillMount()` & stored via `setState()`.  

No worries here though since it isn't standard.

Reply:
It's just not needed on standalone pages and adds complexity. It's also easy to get wrong when you start generating dynamic values and can make test code harder to write.